practice/src/js/services/baseService.js

```diff
@@ -24,6 +24,34 @@ class BaseService {
     return await this.httpClient.delete(`${this.endpoint}/${id}`);
   }
 
+  /**
+   * Sorts the items based on the specified field and order.
+   * @param {string} field - The field to sort the items by.
+   * @param {string} [orderBy=''] - The order by which to sort the items.
+   * @returns {Promise<Object[]>} - A promise resolving to an array of sorted items.
+   */
+  async sort(field, orderBy = '') {
+    const items = (await this.getAll()).slice();
```

Comment:
Does the mock API support sorting?

Reply:
I am not sure why do we need `.slice()`? can you explain?

Reply:
i don't want to modify the original array returned by getAll(), so i use slice(). to create a copy of it. but when I look back, I see it's redundant here, so I'll fix it